src/react/useBaseQuery.js

```diff
@@ -41,15 +45,19 @@ export function useBaseQuery(queryKey, config = {}) {
     instanceRef.current.run()
   }, [enabledBool, query])
 
-  const queryState = query.state
+  // Get hook specific query info
+  const queryInfo = getQueryInfo(query)
+
+  // Handle suspense
+  handleSuspense(queryInfo)
 
-  const queryInfoRef = React.useRef({})
+  return queryInfo
+}
 
-  Object.assign(queryInfoRef.current, {
```

Comment:
I made a few core changes around here that may make some of this unnecessary. Would you like to see if those changes you pulled in solve the issue without these changes?

Reply:
Which changes are you referring to? PS: I combined the deepEqual and shallowEqual functions to reduce file size.

Reply:
Master has a change that uses a Ref right here to stabilize the return value of useBaseQuery. What I'm getting as is that I think a lot of changes with regards to getBaseQueryInfo and all of the implementation to memoize the return values of the hooks might be able to be simplified. But I'm keeping an open mind. I need to look through the code again today.

Reply:
Ah yes I reverted that change (you can find the reasoning for that in https://github.com/tannerlinsley/react-query/pull/754#issuecomment-659252841). But I can also remove the `getBaseQueryInfo` part if you want. It is not needed for memoization but it did allow to move the handling of suspense into the base query hook. Maybe it would be an idea to move the pagination behaviour into the core and enable it with `config.paginated = true` for example just like the infinite query? Then the specific hooks would only do some configuration and all of the behaviour would be in the core or base query hook.